src/components/card/card.component.spec.ts

```diff
@@ -1,28 +1,84 @@
 import { ComponentFixture, TestBed } from '@angular/core/testing';
 import { CardComponent } from './card.component';
+import { MatCardModule } from '@angular/material/card';
+import { TabbedChartsMeta } from '../../datasets/metas/tabbed-charts.meta';
+import { LineChartMeta } from '../../datasets/metas/line-chart.meta';
 
 describe('CardComponent', () => {
-  let fixture: ComponentFixture<CardComponent>;
-  let component: CardComponent;
+  let tabbedFixture: ComponentFixture<CardComponent<TabbedChartsMeta>>;
+  let tabbedComponent: CardComponent<TabbedChartsMeta>;
+  let lineFixture: ComponentFixture<CardComponent<LineChartMeta>>;
+  let lineComponent: CardComponent<LineChartMeta>;
+
+  const testChartMeta1: LineChartMeta = {
+    type: 'line',
+    title: 'testChart1',
+    query: () => [],
+  };
+  const testChartMeta2: LineChartMeta = {
+    type: 'line',
+    title: 'testChart2',
+    query: () => [],
+  };
+  const testTabbedChartsMeta: TabbedChartsMeta = {
+    type: 'tabbed',
+    title: 'testTabbedCharts',
+    metas: [testChartMeta1, testChartMeta2],
+  };
 
   beforeEach(() => {
     TestBed.configureTestingModule({
       declarations: [
         CardComponent,
       ],
+      imports: [
+        MatCardModule
+      ]
     });
-    fixture = TestBed.createComponent(CardComponent);
-    component = fixture.componentInstance;
+    tabbedFixture = TestBed.createComponent<CardComponent<TabbedChartsMeta>>(CardComponent);
+    tabbedComponent = tabbedFixture.componentInstance;
+    lineFixture = TestBed.createComponent<CardComponent<LineChartMeta>>(CardComponent);
+    lineComponent = lineFixture.componentInstance;
   });
 
   it('should instantiate.', () => {
-    expect(component).toBeInstanceOf(CardComponent);
+    expect(tabbedComponent).toBeInstanceOf(CardComponent);
+    expect(lineComponent).toBeInstanceOf(CardComponent);
   });
 
-  it('should init the measure name.', () => {
-    const measureNames = ['a', 'b', 'c'];
-    component.measureNames = measureNames;
-    component.ngOnInit();
-    expect(measureNames.includes(component.currentMeasureName)).toBeTrue();
+  it('should set current chart title when init.', () => {
+    lineComponent.meta = testChartMeta1;
+    lineComponent.ngOnInit();
+    expect(lineComponent.currentChart.title).toBe(testChartMeta1.title);
+
+    tabbedComponent.meta = testTabbedChartsMeta;
+    tabbedComponent.ngOnInit();
+    expect(tabbedComponent.currentChart.title).toBe(testChartMeta1.title);
+  });
+
+  it('should return whether input meta is tabbed charts or not.', () => {
```

Comment:
Would be better to test the user facing behavior instead of part of the implementation (i.e. if the tabs div is present in the DOM)

Reply:
Then do we need to keep these unit tests on return values?

Reply:
Moved to #54 . Will be resolved in separate PRs.